src/bun.js/node/types.zig

```diff
@@ -2107,16 +2107,18 @@ pub const Process = struct {
         }
     }
 
-    pub fn exit(globalObject: *JSC.JSGlobalObject, code: u8) callconv(.C) void {
+    extern fn Bun__setExitCode(vm: *JSC.VirtualMachine, code: u8, explicit: bool) void;
+    pub fn exit(globalObject: *JSC.JSGlobalObject, code: u8, explicit: bool) callconv(.C) void {
         var vm = globalObject.bunVM();
+        Bun__setExitCode(vm, code, explicit);
+
         if (vm.worker) |worker| {
-            vm.exit_handler.exit_code = code;
             worker.requestTerminate();
             return;
         }
 
-        vm.exit_handler.exit_code = code;
         vm.onExit();
```

Comment:
Instead of relying on vm.unhandled_error_counter,  let's create an unhandled rejection scope.
```
const scope = vm.unhandledRejectionScope();